ci/copy_doxygen_files.sh

```diff
@@ -0,0 +1,31 @@
+#!/bin/sh
+
+# Prepare the source tree for Doxygen, see .github/workflows/doxygen.yml
+
+set -ex
+
+# Load the list of modified files.
+git fetch
+git branch -lr
+FILES=$(git diff --name-only origin/master... src/include)
+
+# Create a new temp directory.
+mkdir src/include2
+
+# Copy modified files into it.
+for FILE in $FILES
+do
+    cp "$FILE" ./src/include2
+done
```

Comment:
Then this can just be a single `cp` operation:

```suggestion
cp "${FILES[@]}" ./src/include2
```